### PR TITLE
Update 0.13.2 (hashes)

### PR DESCRIPTION
**Pull request by kirugan**

### CODE REVIEW

File: blockchain/blockchain.go [hunk 1]
```diff
@@ -340,6 +340,7 @@ func (b *Blockchain) Store(block *core.Block, blockCommitments *core.BlockCommit
 		if err := verifyBlock(txn, block); err != nil {
 			return err
 		}
+		return errors.New("temp error")
```
Comment (kirugan) [inline blockchain/blockchain.go:343]: I'll remove it once local sync is correct

File: core/transaction.go [hunk 2]
```diff
@@ -673,6 +665,28 @@ func transactionCommitment(transactions []Transaction, protocolVersion string) (
 	})
 }
 
+func TransactionCommitmentPoseidon(transactions []Transaction) (*felt.Felt, error) {
```
Comment (rianhughes) [inline core/transaction.go:668]: Does this need to be exported? It's tested indirectly via block hash tests right?

File: core/block_test.go [hunk 3]
```diff
@@ -238,12 +240,85 @@ func TestBlockHash(t *testing.T) {
 			"transaction hash (%v) at index: %v does not match receipt's hash (%v)",
 			mainnetBlock1.Transactions[1].Hash().String(), 1,
 			mainnetBlock1.Receipts[1].TransactionHash)
-		commitments, err := core.VerifyBlockHash(mainnetBlock1, &utils.Mainnet)
+		commitments, err := core.VerifyBlockHash(mainnetBlock1, &utils.Mainnet, nil)
 		assert.EqualError(t, err, expectedErr)
 		assert.Nil(t, commitments)
 	})
 }
```
Comment (rianhughes) [inline core/block_test.go:248]: Aneeques PR aims to remove these, so I'll neglect reviewing these specific changes

File: core/block_test.go [hunk 4]
```diff
@@ -257,3 +332,24 @@ func TestBlockHashP2P(t *testing.T) {
 		assert.EqualError(t, err, "block.SequencerAddress is nil")
 	})
 }
+
```
Comment (rianhughes) [inline core/block_test.go:335]: I'll also assume these will be removed

File: core/state_update.go [hunk 5]
```diff
@@ -50,6 +50,41 @@ func (d *StateDiff) Length() uint64 {
 	return uint64(length)
 }
 
+func (d *StateDiff) Hash() *felt.Felt {
```
Comment (rianhughes) [inline core/state_update.go:53]: Aneeques changes also override these, so I'll neglect this change

File: core/transaction.go [hunk 6]
```diff
@@ -117,18 +118,9 @@ type BuiltinInstanceCounter struct {
...
-	Events             []*Event
-	ExecutionResources *ExecutionResources
-	L1ToL2Message      *L1ToL2Message
-	L2ToL1Message      []*L2ToL1Message
-	TransactionHash    *felt.Felt
-	Reverted           bool
-	RevertReason       string
+	AddMod       uint64
+	MulMod       uint64
+	RangeCheck96 uint64
```
Comment (rianhughes) [inline core/transaction.go:123]: Why was TransactionReceipt removed?

File: core/transaction_test.go [hunk 7]
```diff
@@ -350,3 +350,20 @@ func TestDeclareV0TransactionHash(t *testing.T) {
 	require.NoError(t, err)
 	assert.Equal(t, expectedHash, got)
 }
+
+func TestTransactionCommitmentPoseidon(t *testing.T) {
```
Comment (rianhughes) [inline core/transaction_test.go:354]: Assuming this will be removed when Aneeques PR is merged

**Review by IronGauntlets — APPROVED** [review]

File: core/block_test.go [hunk 8]
```diff
@@ -238,12 +238,37 @@ func TestBlockHash(t *testing.T) {
...
 			mainnetBlock1.Transactions[1].Hash().String(), 1,
 			mainnetBlock1.Receipts[1].TransactionHash)
-		commitments, err := core.VerifyBlockHash(mainnetBlock1, &utils.Mainnet)
+		commitments, err := core.VerifyBlockHash(mainnetBlock1, &utils.Mainnet, nil)
 		assert.EqualError(t, err, expectedErr)
 		assert.Nil(t, commitments)
 	})
 }
 
+func Test0132BlocHash(t *testing.T) {
```
Comment (kirugan) [inline core/block_test.go:247]: seems like it meant to be Test0132Bloc**k**Hash

Comment (kirugan) [reply core/block_test.go]: I pushed the fix.

**Review by rianhughes — APPROVED** [review]

**Pull request merged**